src/bundler/bundle_v2.zig

```diff
@@ -742,49 +753,66 @@ pub const BundleV2 = struct {
-        generator.linker.options.public_path = bundler.options.public_path;
-        generator.linker.options.target = bundler.options.target;
-
-        var pool = try generator.graph.allocator.create(ThreadPool);
+        this.linker.graph.allocator = this.graph.heap.allocator();
+        this.graph.allocator = this.linker.graph.allocator;
+        this.bundler.allocator = this.graph.allocator;
+        this.bundler.resolver.allocator = this.graph.allocator;
+        this.bundler.linker.allocator = this.graph.allocator;
+        this.bundler.log.msgs.allocator = this.graph.allocator;
+        this.bundler.log.clone_line_text = true;
+
+        // We don't expose an option to disable this. Kit requires tree-shaking
+        // disabled since every export is always referenced in case a future
+        // module depends on a previously unused export.
+        //
+        // TODO: tree-shaking could be applied still?
+        if (this.bundler.options.output_format == .internal_kit_dev) {
+            this.bundler.options.tree_shaking = false;
+            this.bundler.resolver.opts.tree_shaking = false;
+        } else {
+            this.bundler.options.tree_shaking = true;
+            this.bundler.resolver.opts.tree_shaking = true;
+        }
+
+        this.linker.graph.bundler_graph = &this.graph;
+        this.linker.resolver = &this.bundler.resolver;
+        this.linker.graph.code_splitting = bundler.options.code_splitting;
+        this.graph.code_splitting = bundler.options.code_splitting;
+
+        this.linker.options.minify_syntax = bundler.options.minify_syntax;
+        this.linker.options.minify_identifiers = bundler.options.minify_identifiers;
+        this.linker.options.minify_whitespace = bundler.options.minify_whitespace;
+        this.linker.options.emit_dce_annotations = bundler.options.emit_dce_annotations;
+        this.linker.options.ignore_dce_annotations = bundler.options.ignore_dce_annotations;
+
+        this.linker.options.source_maps = bundler.options.source_map;
+        this.linker.options.tree_shaking = bundler.options.tree_shaking;
+        this.linker.options.public_path = bundler.options.public_path;
+        this.linker.options.target = bundler.options.target;
+        this.linker.options.output_format = bundler.options.output_format;
+
+        var pool = try this.graph.allocator.create(ThreadPool);
         if (enable_reloading) {
-            Watcher.enableHotModuleReloading(generator);
+            Watcher.enableHotModuleReloading(this);
         }
         // errdefer pool.destroy();
-        errdefer generator.graph.heap.deinit();
+        errdefer this.graph.heap.deinit();
 
         pool.* = ThreadPool{};
-        generator.graph.pool = pool;
+        this.graph.pool = pool;
         try pool.start(
             this,
             thread_pool,
         );
 
-        return generator;
+        // sanity checks for kit
+        if (this.bundler.options.output_format == .internal_kit_dev) {
```

Comment:
should this be wrapped in `if (isDebug)`?

Reply:
we can, but i don't want the bundler to ever continue if these options are misconfigured. these checks are not expensive.